### DISCUSSION TITLE
Quickly hovering over `ToolbarButton`s that have hotkeys causes lag during score upgrade process

### DISCUSSION THREAD

**Discussion by Stedoss** [ppy/osu#26018] [Q&A] [answered]
Feel free to ignore this, since it's super insignificant. I noticed this a while back, but was too lazy to recreate it in a dev instance.

I think this will happen during any realm-heavy background operation since on every hover it fetches the current keybindings from realm:

https://github.com/ppy/osu/blob/340227a06d65c6b8bb1edf2e4834ab04d7f809ac/osu.Game/Overlays/Toolbar/ToolbarButton.cs#L204

Exaggerated mouse flailing video:

https://github.com/ppy/osu/assets/29103029/858286bc-8820-40fe-8c24-2af9e9a6c459

Logs:
[compressed-logs.zip](https://github.com/ppy/osu/files/13745806/compressed-logs.zip)

**Comment by bdach** [comment]
This is presumably happening because showing the tooltip text causes a query to the realm db to retrieve the key combination for each tooltip, and the database in the mean time is bombarded with writes from the processing, causing long blocking reads.

**Reply by bdach** [reply, marked as answer]
https://github.com/ppy/osu/pull/26086
